internal/oid/algorithm.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oid
+
+import (
+	"crypto/x509"
+	"encoding/asn1"
+)
+
+// ToSignatureAlgorithm converts ASN.1 digest and signature algorithm
+// identifiers to golang signature algorithms.
+func ToSignatureAlgorithm(digestAlg, sigAlg asn1.ObjectIdentifier) x509.SignatureAlgorithm {
+	switch {
+	case RSA.Equal(sigAlg):
+		switch {
+		case SHA256.Equal(digestAlg):
+			return x509.SHA256WithRSA
+		case SHA384.Equal(digestAlg):
+			return x509.SHA384WithRSA
+		case SHA512.Equal(digestAlg):
+			return x509.SHA512WithRSA
+		}
```

Comment:
@priteshbandi Please skip the review of files in the `internal` folder, because they will be reviewed and merged in the dependent PR #17. These files are here just for compile and integration tests.